### PR TITLE
feat(Views): add connectDecorator for react and inferno

### PR DESCRIPTION
**Pull request by etienne-dldc**
add a `connectDecorator` export to `cerebral/react` and `cerebral/inferno` that work just like
`connect` but instead of passing the component as last argument, it return a function that take the
component as unique argument meaning you can use it as a decorator.

ISSUES CLOSED: #468

### CODE REVIEW

**Comment by Guria** [comment]
Thanks, great PR.

@christianalfoni what do you think about export name for decorator? connectDecorator vs just decorator?

**Comment by christianalfoni** [comment]
Good question, I think `decorator` as it is exported from `cerebral/react`... 

```js
import {decorator as connect} from 'cerebral/react'
```

Though no strong opinion :)

**Comment by Guria** [comment]
@etienne-dldc please update it to use decorator as export name

File: packages/cerebral-debugger/package.json [hunk 1]
```diff
@@ -25,10 +25,10 @@
     "prepublish": "npm run build"
   },
   "dependencies": {
-    "cerebral": "next",
-    "cerebral-module-useragent": "next",
-    "color": "^0.11.4",
+    "cerebral": "^2.0.0-0-alpha.36dab802",
+    "cerebral-module-useragent": "^1.0.0-0-alpha.36dab802",
```
Comment (edgesoft) [inline packages/cerebral-debugger/package.json]: @Guria Same issue here.

File: packages/cerebral-module-useragent/package.json [hunk 2]
```diff
@@ -21,9 +21,9 @@
   "bugs": {
     "url": "https://github.com/cerebral/cerebral/issues"
   },
-  "homepage":
-    "https://github.com/cerebral/cerebral/tree/master/packages/cerebral-module-useragent#readme",
+  "homepage": "https://github.com/cerebral/cerebral/tree/master/packages/cerebral-module-useragent#readme",
   "dependencies": {
+    "cerebral": "^2.0.0-0-alpha.36dab802",
```
Comment (edgesoft) [inline packages/cerebral-module-useragent/package.json]: @Guria  And here

File: packages/cerebral-provider-firebase/package.json [hunk 3]
```diff
@@ -20,9 +20,9 @@
   "bugs": {
     "url": "https://github.com/cerebral/cerebral/issues"
   },
-  "homepage":
-    "https://github.com/cerebral/cerebral/tree/master/packages/cerebral-provider-firebase#readme",
+  "homepage": "https://github.com/cerebral/cerebral/tree/master/packages/cerebral-provider-firebase#readme",
   "dependencies": {
+    "cerebral": "^2.0.0-0-alpha.36dab802",
```
Comment (edgesoft) [inline packages/cerebral-provider-firebase/package.json]: Problem with next.

File: packages/cerebral-provider-http/package.json [hunk 4]
```diff
@@ -28,5 +28,9 @@
   "devDependencies": {
     "cerebral": "next",
     "xhr-mock": "https://github.com/christianalfoni/xhr-mock.git"
+  },
+  "dependencies": {
+    "cerebral": "^2.0.0-0-alpha.36dab802",
```
Comment (edgesoft) [inline packages/cerebral-provider-http/package.json]: Here as well

File: packages/cerebral/package.json [hunk 5]
```diff
@@ -18,7 +18,7 @@
   "homepage": "http://cerebral.github.io/cerebral-website",
   "dependencies": {
     "eventemitter3": "^2.0.2",
-    "function-tree": "next"
+    "function-tree": "^0.5.0-0-alpha.36dab802"
```
Comment (edgesoft) [inline packages/cerebral/package.json]: Here as well

File: packages/cerebral-router/package.json [hunk 6]
```diff
@@ -18,6 +18,7 @@
   "homepage": "https://cerebral.github.io/cerebral-website/api/08_router.html",
   "dependencies": {
     "addressbar": "^1.0.3",
+    "cerebral": "^2.0.0-0-alpha.36dab802",
```
Comment (edgesoft) [inline packages/cerebral-router/package.json]: here as well

File: packages/demo/package.json [hunk 7]
```diff
@@ -14,8 +14,8 @@
   },
   "dependencies": {
     "bulma": "^0.2.3",
-    "cerebral": "next",
-    "cerebral-router": "next",
+    "cerebral": "^2.0.0-0-alpha.36dab802",
```
Comment (edgesoft) [inline packages/demo/package.json]: not next

Comment (edgesoft) [inline packages/demo/package.json]: not next

File: packages/demo/package.json [hunk 8]
```diff
@@ -14,8 +14,8 @@
...
+    "cerebral-router": "^1.0.0-0-alpha.36dab802",
```
Comment (edgesoft) [inline packages/demo/package.json]: not next

File: packages/function-tree-demos/package.json [hunk 9]
```diff
@@ -12,7 +12,7 @@
     "axios": "^0.15.2",
     "body-parser": "^1.15.2",
     "express": "^4.14.0",
-    "function-tree": "next",
+    "function-tree": "^0.5.0-0-alpha.36dab802",
```
Comment (edgesoft) [inline packages/function-tree-demos/package.json]: not next

File: packages/function-tree-firebase-demo/package.json [hunk 10]
```diff
@@ -4,7 +4,7 @@
   "scripts": {},
   "dependencies": {
     "firebase": "^3.5.1",
-    "function-tree": "next"
+    "function-tree": "^0.5.0-0-alpha.36dab802"
```
Comment (edgesoft) [inline packages/function-tree-firebase-demo/package.json]: not next

File: packages/todomvc/package.json [hunk 11]
```diff
@@ -11,8 +11,8 @@
     "react-scripts": "0.7.0"
   },
   "dependencies": {
-    "cerebral": "next",
-    "cerebral-router": "next",
+    "cerebral": "^2.0.0-0-alpha.36dab802",
```
Comment (edgesoft) [inline packages/todomvc/package.json]: not next

File: packages/todomvc/package.json [hunk 12]
```diff
@@ -11,8 +11,8 @@
...
+    "cerebral-router": "^1.0.0-0-alpha.36dab802",
```
Comment (edgesoft) [inline packages/todomvc/package.json]: not next

File: packages/tutorial/package.json [hunk 13]
```diff
@@ -15,9 +15,9 @@
     "shelljs": "^0.7.5"
   },
   "dependencies": {
-    "cerebral": "next",
-    "cerebral-provider-http": "next",
-    "cerebral-router": "next",
+    "cerebral": "^2.0.0-0-alpha.36dab802",
```
Comment (edgesoft) [inline packages/tutorial/package.json]: not next

File: packages/tutorial/package.json [hunk 14]
```diff
@@ -15,9 +15,9 @@
...
+    "cerebral-provider-http": "^1.0.0-0-alpha.36dab802",
```
Comment (edgesoft) [inline packages/tutorial/package.json]: not next

File: packages/tutorial/package.json [hunk 15]
```diff
@@ -15,9 +15,9 @@
...
+    "cerebral-router": "^1.0.0-0-alpha.36dab802",
```
Comment (edgesoft) [inline packages/tutorial/package.json]: not next

**Comment by edgesoft** [comment]
@Guria @christianalfoni What do we do with the next thing?

**Comment by Guria** [comment]
must be kept as "next" string in package.json

**Comment by edgesoft** [comment]
@etienne-dldc Can you update package.json to next? Sorry about all this :) but it's WIP to 2.0

**Comment by edgesoft** [comment]
@etienne-dldc Great PR

**Comment by etienne-dldc** [comment]
@edgesoft Yes, I will fix this, sorry for the mess :/

**Comment by edgesoft** [comment]
@etienne-dldc Great work! We haven't been able to be in sync with migration docs 👍

**Comment by etienne-dldc** [comment]
It should be ok now, `package.json` are not changed by the PR.

**Comment by edgesoft** [comment]
Looks great!

**Comment by Guria** [comment]
Ouch. Still Smith wrong here. Let me review it tomorrow

**Pull request merged**